src/shared/nested-dropdown-list/types.ts

```diff
@@ -0,0 +1,84 @@
+import { Option } from "../../input-nested-select";
+
+export type TruncateType = "middle" | "end";
+export type ItemsLoadStateType = "loading" | "fail" | "success";
+export type Mode = "default" | "expand" | "collapse";
+
+export interface ListItemRenderArgs {
+    selected: boolean;
+}
+
+export interface DropdownDisplayProps<V1, V2, V3> {
+    /** Control the dropdown list when it is first opened */
+    mode?: Mode;
+    /** Expand the recursively selected key in dropdown list when it is first opened */
+    selectedKeyPath: string[];
+    /** Function to derive value from an item */
+    valueExtractor?: ((item: V1 | V2 | V3) => V1 | V2 | V3) | undefined;
```

Comment:
ohh.. ok. removed it.

Reply:
yup, it was useful for the original Select because anything in the option could be the value. but since we explicitly have `value`, the consumer should be setting the value they need anyway